core-bundle/src/DependencyInjection/Compiler/AddNativeTransportFactoryPass.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Contao\CoreBundle\Util\PackageUtil;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Mailer\Transport\NativeTransportFactory;
+
+/**
+ * @internal
+ */
+class AddNativeTransportFactoryPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if ($container->hasDefinition('mailer.transport_factory.native') || !$this->supportsNativeMailer()) {
+            return;
+        }
+
+        $definition = new ChildDefinition('mailer.transport_factory.abstract');
+        $definition
+            ->setClass(NativeTransportFactory::class)
+            ->addTag('mailer.transport_factory')
+        ;
+
+        $container->setDefinition('mailer.transport_factory.native', $definition);
+    }
+
+    private function supportsNativeMailer(): bool
+    {
+        return version_compare(PackageUtil::getNormalizedVersion('symfony/mailer'), '5.2.0', '>=');
```

Comment:
I think it would be better to use `class_exists(NativeTransportFactory::class)` here.